doc/src/manual/arrays.md

```diff
@@ -424,8 +424,12 @@ subsections, and arrays of booleans to select elements at their `true` indices.
 
 If `X` is an array, it must have the same number of elements as the product of the lengths of
```

Comment:
Let's also invert the If-clause here and tighten up this description while we're at it.  Maybe something like:

> If any index `I_k` selects more than one location, then the right hand side `X` must be an array with the same shape* as the result of indexing `A[I_1, I_2, ..., I_n]` or a vector with the same number of elements.

This could lead to a slightly different wording than what you proposed:

> The dotted assignment operator `.=` can be used to broadcast the assignment of fewer elements in `X` across a compatible shape selected by the indices `[I_1, I_2, …, I_n]`.

What do you think?

* That's still not quite true as we also allow adding or removing singleton dimensions to make the shapes match, but I find that extra condition quite confusing to explain.